mongo/datadog_checks/mongo/mongo.py

```diff
@@ -777,80 +777,81 @@ def total_seconds(td):
-        try:
-            data = {}
-            dbnames = []
-
-            replSet = admindb.command('replSetGetStatus')
-            if replSet:
-                primary = None
-                current = None
-
-                # need a new connection to deal with replica sets
-                setname = replSet.get('set')
-                cli_rs = pymongo.mongo_client.MongoClient(
-                    server,
-                    socketTimeoutMS=timeout,
-                    connectTimeoutMS=timeout,
-                    serverSelectionTimeoutMS=timeout,
-                    replicaset=setname,
-                    read_preference=pymongo.ReadPreference.NEAREST,
-                    **ssl_params)
-
-                if do_auth:
-                    if auth_source:
-                        self._authenticate(cli_rs[auth_source], username, password, use_x509, server, service_check_tags)
-                    else:
-                        self._authenticate(cli_rs[db_name], username, password, use_x509, server, service_check_tags)
-
-                # Replication set information
-                replset_name = replSet['set']
-                replset_state = self.get_state_name(replSet['myState']).lower()
-
-                tags.extend([
-                    u"replset_name:{0}".format(replset_name),
-                    u"replset_state:{0}".format(replset_state),
-                ])
-
-                # Find nodes: master and current node (ourself)
-                for member in replSet.get('members'):
-                    if member.get('self'):
-                        current = member
-                    if int(member.get('state')) == 1:
-                        primary = member
-
-                # Compute a lag time
-                if current is not None and primary is not None:
-                    if 'optimeDate' in primary and 'optimeDate' in current:
-                        lag = primary['optimeDate'] - current['optimeDate']
-                        data['replicationLag'] = total_seconds(lag)
-
-                if current is not None:
-                    data['health'] = current['health']
-
-                data['state'] = replSet['myState']
-
-                if current is not None:
-                    total = 0.0
-                    cfg = cli_rs['local']['system.replset'].find_one()
-                    for member in cfg.get('members'):
-                        total += member.get('votes', 1)
-                        if member['_id'] == current['_id']:
-                            data['votes'] = member.get('votes', 1)
-                    data['voteFraction'] = data['votes'] / total
-
-                status['replSet'] = data
-
-                # Submit events
-                self._report_replica_set_state(
-                    data['state'], clean_server_name, replset_name, self.agentConfig
-                )
+        if not instance.get('disable_replica_check'):
```

Comment:
We may want to use the self._is_affirmative function here so we can handle the variations of `Yes`, `yes`, `no`, `No`, etc.